vue-components/src/components/Checkbox.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+	<span class="wikit wikit-checkbox">
+		<input
+			type="checkbox"
+			:id="id"
+			class="wikit-checkbox__box"
+			@change="$emit( 'input', $event.target.checked )"
+			:value="nativeValue"
+			:checked="value"
+			:disabled="disabled"
+		>
+		<label
+			class="wikit-checkbox__label"
+			:for="id"
+		>{{ label }}</label>
+	</span>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import generateUid from '@/components/util/generateUid';
+
+export default Vue.extend( {
+	name: 'Checkbox',
+	props: {
+		value: {
+			required: true,
+			type: Boolean,
+		},
+		nativeValue: {
+			default: null,
+			type: String,
+		},
+		label: {
+			required: true,
+			type: String,
+		},
+		disabled: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	data() {
+		return {
+			// https://github.com/vuejs/vue/issues/5886
+			id: generateUid( 'wikit-Checkbox' ),
+		};
+	},
+} );
+</script>
+
+<style lang="scss">
+$label: '.wikit-checkbox__label';
+
+.wikit-checkbox {
+	&__label {
+		display: flex;
+		align-items: center;
+
+		// TODO: replace with Label mixin
+		color: $wikit-Label-font-color;
+		font-family: $wikit-Label-font-family;
+		font-size: $wikit-Label-font-size;
+		font-weight: $wikit-Label-font-weight;
+		line-height: $wikit-Label-line-height;
+
+		&::before {
+			width: $wikit-Checkbox-input-size;
+			height: $wikit-Checkbox-input-size;
+
+			border-width: $wikit-Checkbox-input-border-width;
+			border-color: $wikit-Checkbox-input-border-color;
+			border-style: $wikit-Checkbox-input-border-style;
+			border-radius: $wikit-Checkbox-input-border-radius;
+			background-color: $wikit-Checkbox-input-background-color; /* Workaround Firefox e.a. */
+
+			content: ' ';
+			margin-inline-end: $wikit-Checkbox-input-right-spacing;
+			display: flex;
+			transition-property: $wikit-Checkbox-input-transition-property;
+			transition-duration: $wikit-Checkbox-input-transition-duration;
+			transition-timing-function: $wikit-Checkbox-input-transition-timing-function;
+		}
+
+		&:hover {
+			&::before {
+				border-color: $wikit-Checkbox-input-hover-border-color;
+			}
+		}
+
+		&:active {
+			&::before {
+				border-color: $wikit-Checkbox-input-active-border-color;
+				background-color: $wikit-Checkbox-input-active-background-color;
+			}
+		}
+	}
+
+	&__box {/* see: https://www.w3schools.com/howto/howto_css_custom_checkbox.asp */
```

Comment:
Would it make sense to call this "input" instead of "box", this way it would match the tokens and be more inline with the usual element markup? Same naming would apply to radio buttons. Wdyt?

Reply:
Sounds like a good suggestion to me :+1: